MyCapytain/resolvers/capitains/local.py

```diff
@@ -0,0 +1,497 @@
+        else:
+            # Perhaps there is a better exception to raise here?
+            raise(UnknownObjectError('{} is not a readable object'.format(str(identifier))))
+
+        if os.path.isfile(text.path):
+            with io.open(text.path) as __xml__:
+                resource = self.classes["text"](urn=identifier, resource=self.xmlparse(__xml__))
+        else:
+            # Passing None to the functions that call __getText__ results in a not very helpful AttributeError
+            # A more informative error should be raised here.
+            resource = None
+            self.logger.warning('The file {} is mentioned in the metadata but does not exist'.format(text.path))
+
+        return resource, text
+
+    def __getTextMetadata__(self,
+                            urn=None, page=None, limit=None,
+                            lang=None, category=None, pagination=False
+                            ):
+        """ Retrieve a slice of the inventory filtered by given arguments
+        :param urn: Partial URN to use to filter out resources
+        :type urn: str
+        :param page: Page to show
+        :type page: int
+        :param limit: Item Per Page
+        :type limit: int
+        :param inventory: Inventory name
+        :type inventory: str
+        :param lang: Language to filter on
+        :type lang: str
+        :param category: Type of elements to show
+        :type category: str
+        :param pagination: Activate pagination
+        :type pagination: bool
+        :return: ([Matches], Page, Count)
+        :rtype: ([CtsTextMetadata], int, int)
+        """
+        if urn is not None:
+            collection = self.inventory[urn].readableDescendants or [self.inventory[urn]]
+        else:
+            collection = self.inventory.readableDescendants
+
+        matches = [
+            text
+            for text in collection
+            if
+            (lang is None or (lang is not None and lang == text.lang)) and
+            (text.citation is not None) and
+            # I am not sure that this categorization is useful since the type is not longer a strictly controlled
+            # vocabulary.
+            # It might be possible if the resolver built a list of the types as it was ingesting the texts and then
+            # checked to see if the category parameter is a member of this list.
+            (
+                category not in ["cts:edition", "cts:translation", "cts:commentary"] or
+                (category in ["cts:edition", "cts:translation", "cts:commentary"]
+                 and category.lower() == text.subtype.lower())
+            )
+        ]
+        if pagination:
+            start_index, end_index, page, count = type(self).pagination(page, limit, len(matches))
+        else:
+            start_index, end_index, page, count = None, None, 0, len(matches)
+
+        return matches[start_index:end_index], page, count
+
+    @staticmethod
+    def pagination(page, limit, length):
```

Comment:
Skrew that ?

Reply:
This is still used in the `._get_text_metadata` function. So I will leave it for now.